src/github/summary.ts

```diff
@@ -81,18 +92,34 @@ function getEventLink(): Link {
     }
   }
 
-  if (isPushEvent(context) && context.payload.head_commit) {
+  if (isPushEvent(context)) {
     const commit = context.payload.head_commit
 
+    if (!commit) {
+      throw new Error('Unexpected push event payload (undefined head_commit)')
+    }
+
+    return {
+      text: getEventLinkText(commit.message),
+      url: commit.url
+    }
+  }
+
+  if (isScheduleEvent(context)) {
+    const commit = (
+      await octokit.rest.repos.getCommit({
+        ...context.repo,
+        ref: context.sha
+      })
+    ).data.commit
```

Comment:
The `schedule` event payload doesn't have much – mostly just repository information – so we have to fetch the commit.